crates/rpc/rpc/src/eth/api/pending_block.rs

```diff
@@ -43,13 +49,44 @@ impl PendingBlockEnv {
         let mut db = State::builder().with_database(Box::new(state)).with_bundle_update().build();
 
         let mut cumulative_gas_used = 0;
+        let mut sum_blob_gas_used = 0;
         let block_gas_limit: u64 = block_env.gas_limit.try_into().unwrap_or(u64::MAX);
         let base_fee = block_env.basefee.to::<u64>();
         let block_number = block_env.number.to::<u64>();
 
         let mut executed_txs = Vec::new();
         let mut best_txs = pool.best_transactions_with_base_fee(base_fee);
 
+        let (withdrawals, withdrawals_root) = match origin.clone() {
+            PendingBlockEnvOrigin::ActualPending(block) => {
+                (block.clone().withdrawals, block.withdrawals_root)
+            }
+            PendingBlockEnvOrigin::DerivedFromLatest(_) => (None, None),
+        };
+
+        let parent_beacon_block_root = if origin.is_actual_pending() {
+            origin.header().parent_beacon_block_root
+        } else {
+            None
+        };
+
+        let chain_spec = ChainSpec::default(); // mainnet
+
+        // apply eip-4788 pre block contract call
+        pre_block_beacon_root_contract_call(
```

Comment:
we only need to call this if actually pending, right?

so can move into if branch?

Reply:
yes I think so too. Because if it's not actually pending this contract call has already been called before